include/cgimap/request_context.hpp

```diff
@@ -0,0 +1,37 @@
+/**
+ * SPDX-License-Identifier: GPL-2.0-only
+ *
+ * This file is part of openstreetmap-cgimap (https://github.com/zerebubuth/openstreetmap-cgimap/).
+ *
+ * Copyright (C) 2009-2024 by the CGImap developer community.
+ * For a full list of authors see the git log.
+ */
+
+#ifndef REQUEST_CONTEXT_HPP
+#define REQUEST_CONTEXT_HPP
+
+#include "cgimap/types.hpp"
+
+#include <optional>
+#include <set>
+#include <chrono>
+
+struct request;
+
+struct UserInfo
+{
+    osm_user_id_t id = {};
+    std::set<osm_user_role_t> user_roles = {};
+    bool allow_api_write = false;
+
+    bool has_role(osm_user_role_t role) const { return user_roles.count(role) > 0; }
+};
+
+struct RequestContext
+{
+    request* req = nullptr;
```

Comment:
When checking the new signature for process_post_put_request, I was wondering why we're passing both request and RequestContent to that method. Then I realized that RequestContext.req is not being used at all. Is this intended as is? 

For the JSON upload I was looking to have access to the request headers inside changeset_upload_handler, which doesn't seem to be possible at the moment, since the instance variable is not being set.

Reply:
I could swear I set that :thinking: 
Either it got lost somewhere during refactoring or it was just my imagination. 
I'll change the function to use just the RequestContext as well.